### DISCUSSION TITLE
Is there possibility to run command before compression?

### DISCUSSION THREAD

**Discussion by Krzyrok** [dotnet/aspnetcore#25447] [General] [answered]
I'd like to run my script during build and publish, which will modify `blazor.webassembly.js` but before compression is started (so before `blazor.webassembly.js.br` and `blazor.webassembly.js.gz` are generated). Is there any Target before/after I can run my script (adding correct entry in the csproj file)?
Currently I have: 
```
  <Target Name="ReplacementDuringBuild" BeforeTargets="Build">
    <Exec WorkingDirectory="$(MyDirWithScripts)" Command="node replace.js --output=$(MSBuildThisFileDirectory)$(OutDir)" />
  </Target>

  <Target Name="ReplacementDuringPublish" AfterTargets="AfterPublish">
    <Exec WorkingDirectory="$(MyDirWithScripts)" Command="node replace.js --output=$(MSBuildThisFileDirectory)$(PublishDir)" />
  </Target>
```
This is workaround (not fully working) for issue described here (there you can find background story): https://github.com/dotnet/aspnetcore/issues/22220#issuecomment-683783971 .
Any suggestion?

**Comment by SteveSandersonMS** [comment, marked as answer]
We don't have a specific MSBuild target for this purpose. You could look at the framework sources and try to find a place that's suitable for plugging in your custom logic, but it's outside the normal range of supported scenarios.

However instead of doing this I would make a different suggestion. I see your goal is to load `blazor.boot.json` from a custom location. You could do that by patching JavaScript's `fetch` function. For example:

```html
<script>
    var originalFetch = window.fetch;
    window.fetch = function(url, options) {
        if (url === '_framework/blazor.boot.json') {
            // Custom logic to return any content you want
            return originalFetch('https://example.com/myCustomUrl/blazor.boot.json', options);
        } else {
            // Use default logic
            return originalFetch.apply(this, arguments);
        }
    };
</script>
<script src="_framework/blazor.webassembly.js"></script>
```

Note: I'm just writing this code from memory - I haven't tested it, so there may be small errors to correct. But I expect you'll get the idea.

**Reply by Krzyrok** [reply]
Yes, I thought about this. I tried to avoid this but probably this is better than modifying files (especially compress them by my own if there is no `Target` for compression)
